src/Driver/PDO/SQLSrv/Statement.php

```diff
@@ -28,11 +28,20 @@ public function __construct(PDOStatement $statement)
      */
     public function bindParam($param, &$variable, $type = ParameterType::STRING, $length = null, $driverOptions = null)
     {
-        if (
-            ($type === ParameterType::LARGE_OBJECT || $type === ParameterType::BINARY)
-            && $driverOptions === null
-        ) {
-            $driverOptions = PDO::SQLSRV_ENCODING_BINARY;
+        switch ($type) {
+            case ParameterType::LARGE_OBJECT:
+            case ParameterType::BINARY:
+                if ($driverOptions === null) {
+                    $driverOptions = PDO::SQLSRV_ENCODING_BINARY;
+                }
+
+                break;
+
+            case ParameterType::ASCII:
+                $type          = PDO::PARAM_STR;
```

Comment:
This is what I was talking about. Look that the merged PR: https://github.com/doctrine/dbal/pull/4274/files#r491219932.

Reply:
Ok, I'll add this change, but not the other one, since it looks more simple the way it is on 3.0.x

Reply:
That other one should stay as is, right.